prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -524,30 +533,30 @@ class Hl7Serializer(
                     if (suppressedFields.contains(hl7Field))
                         return@outputFields
                     if (element.hl7Field != null && element.isTableLookup) {
-                        setComponentForTable(terser, element, hl7Field, report, row, hl7Config)
+                        setComponentForTable(terser, element, hl7Field, hl7Report, row, hl7Config)
                     } else {
-                        setComponent(terser, element, hl7Field, repeat = null, value, report)
+                        setComponent(terser, element, hl7Field, repeat = null, value, hl7Report)
                     }
                 }
             } else if (element.hl7Field == "AOE" && element.type == Element.Type.NUMBER && !suppressAoe) {
                 if (value.isNotBlank()) {
-                    val units = report.getString(row, "${element.name}_units")
-                    val date = report.getString(row, "specimen_collection_date_time") ?: ""
-                    setAOE(terser, element, aoeSequence++, date, value, report, row, units, suppressQst)
+                    val units = hl7Report.getString(row, "${element.name}_units")
+                    val date = hl7Report.getString(row, "specimen_collection_date_time") ?: ""
+                    setAOE(terser, element, aoeSequence++, date, value, hl7Report, row, units, suppressQst)
                 }
             } else if (element.hl7Field == "AOE" && !suppressAoe) {
                 if (value.isNotBlank()) {
-                    val date = report.getString(row, "specimen_collection_date_time") ?: ""
-                    setAOE(terser, element, aoeSequence++, date, value, report, row, suppressQst = suppressQst)
+                    val date = hl7Report.getString(row, "specimen_collection_date_time") ?: ""
+                    setAOE(terser, element, aoeSequence++, date, value, hl7Report, row, suppressQst = suppressQst)
                 } else {
                     // if the value is null but we're defaulting
                     if (hl7Config?.defaultAoeToUnknown == true) {
-                        val date = report.getString(row, "specimen_collection_date_time") ?: ""
-                        setAOE(terser, element, aoeSequence++, date, "UNK", report, row, suppressQst = suppressQst)
+                        val date = hl7Report.getString(row, "specimen_collection_date_time") ?: ""
+                        setAOE(terser, element, aoeSequence++, date, "UNK", hl7Report, row, suppressQst = suppressQst)
                     }
                 }
             } else if (element.hl7Field == "ORC-21-1") {
-                setOrderingFacilityComponent(terser, rawFacilityName = value, useOrderingFacilityName, report, row)
+                setOrderingFacilityComponent(terser, rawFacilityName = value, useOrderingFacilityName, hl7Report, row)
```

Comment:
I added a few lines of code here, earlier so you will have merge conflicts. Just making you aware.